stablehlo/reference/Tensor.cpp

```diff
@@ -50,6 +51,32 @@ int64_t getSizeInBytes(Type type) {
   report_fatal_error(std::move(err));
 }
 
+// Flattens multi-dimensional index 'index' of a tensor to a linearized
+// index into the underlying storage using the dimension strides `strides_`.
+int64_t flattenindex(ArrayRef<int64_t> shape, ArrayRef<int64_t> index) {
```

Comment:
Might be a typo? I think "flattenIndex" reads better, because it's two words. If you're looking for a single word, perhaps consider "linearize"?